(docs): Upddate contribution guidelines around AI usage for PRs/issues (#1598)
modelcontextprotocol/modelcontextprotocol: CONTRIBUTING.md

> was fully authored manually by myself.

Failure to disclose this is first and foremost rude to the human operators on the other end of the pull request,
but it also makes it difficult to determine how much scrutiny to apply to the contribution.

In a perfect world, AI assistance would produce equal or higher quality work than any human. That isn't the world
we live in today, and in most cases where human supervision or expertise is not in the loop, it's generating code
or changes that cannot be reasonably maintained or evolved.

### What we're looking for

When submitting AI-assisted contributions, please ensure they include:

- **Clear disclosure of AI use** - You are transparent about AI use and degree to which you're using it for the contribution
- **Human understanding and testing** - You've personally tested the changes and understand what they do
- **Clear rationale** - You can explain why the change is needed and how it fits within Model Context Protocol goals
- **Concrete evidence** - Include test cases, scenarios, or examples that demonstrate the improvement
- **Your own analysis** - Share your thoughts on the end-to-end experience

### What we'll close

> We reserve the right to close contributions that appear to be:
> 
> - Untested changes submitted without verification
> - Generic suggestions that don't address specific Model Context Protocol needs
> - Bulk submissions that show no human review or understanding

Reviewer: I'm not sure whether these three points apply to the AI-generated submissions I've seen in this repository.

What do you think about something more general?

```suggestion
We reserve the right to close submissions that appear to not follow this disclosure policy.
```
